From 48361b7b89d09fec268d96df22657d0e0224af37 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Sat, 5 Apr 2014 11:05:39 +0100
Subject: [PATCH 177/351] ASoC: florida: Update trigger indicator regardless
 of callback

Currently, we only update the triggered flag if we have a machine driver
callback to inform of the trigger. This flag is useful for other things
so this patch updates it regardless of if a callback is specified.

Change-Id: I56b69b4f488f49e75698e4a60ae3408639f6c4ae
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/florida.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index 60a678d..2c03a37 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -1761,11 +1761,11 @@ static irqreturn_t adsp2_irq(int irq, void *data)
 
 	mutex_lock(&florida->compr_info.lock);
 
-	if (florida->core.arizona->pdata.ez2ctrl_trigger &&
-	    !florida->compr_info.trig &&
+	if (!florida->compr_info.trig &&
 	    florida->core.adsp[2].fw_id == 0x4000d &&
 	    florida->core.adsp[2].running) {
-		florida->core.arizona->pdata.ez2ctrl_trigger();
+		if (florida->core.arizona->pdata.ez2ctrl_trigger)
+			florida->core.arizona->pdata.ez2ctrl_trigger();
 		florida->compr_info.trig = true;
 	}
 
-- 
1.7.9.5

